distro.py

```diff
@@ -1074,7 +1074,71 @@ def _parse_distro_release_content(line):
         return distro_info
 
 
-_distro = LinuxDistribution()
+def _normalize_id(distro_id, table):
+    """ Helper function which normalizes a distro id value. """
+    distro_id = distro_id.lower().replace(' ', '_')
+    return table.get(distro_id, distro_id)
+
+
+class LinuxMintDistribution(LinuxDistribution):
+    """ LinuxMint will have the same /etc/os-release file
+    as it's upstream Ubuntu so we should not use any info
+    from that file except `ID_LIKE`. """
+    def os_release_attr(self, attribute):
+        if attribute == 'id_like':
+            v = super(LinuxMintDistribution, self).os_release_attr(attribute)
+            return v
+        return ''
+
+
+class CloudLinuxDistribution(LinuxDistribution):
+    def id(self):
```

Comment:
i mean both: changing the design and as a side-effect to 'cache'.

```python

def _get_id():
    …

ID = _get_id()

__all__ = ['ID']
```

though i'd argue that no uppercase reads cleaner in the client code: `f'Running on {distro.name}.'`

@nir0s shall i open an issue regarding the design discussion? though i can't promise to layout a detailed proposal soon, i see the opportunity to introduce breaking changes with a new major release. sorry for bloating this up here.